### PR TITLE
feat: request the approval for database migrations

### PR DESCRIPTION
**Pull request by yangby-cryptape**
- docs: fix a typo
- chore: no migration version means the database is uninitialized
  At present, `migration_check` will return `true` if the database is uninitialized.
  But an uninitialized database doesn't have data required to migrate.
  This is important when do `migration_check` in `ckb run`.
- feat: request the approval for database migrations

### CODE REVIEW

File: ckb-bin/src/subcommand/migrate.rs [hunk 1]
```diff
@@ -13,6 +15,28 @@ pub fn migrate(args: MigrateArgs, async_handle: Handle) -> Result<(), ExitCode>
         }
     }
 
+    if !args.force {
+        if atty::is(atty::Stream::Stdin) && atty::is(atty::Stream::Stdout) {
```
Comment (doitian) [inline ckb-bin/src/subcommand/migrate.rs:19]: No need to prompt if `builder.migration_check()` returns true.

**Review by doitian — CHANGES_REQUESTED** [review]

**Comment by doitian** [comment]
Some migrations may be very fast to execute, such as adding a single key into the database. We can let the migration tell whether it is time consuming, and only ask for confirmations in `ckb migration` and disable `ckb run` automatic migration when there’s at least a time consuming migration. However, it could be another PR.

**Review by doitian — APPROVED** [review]

**Review by quake — APPROVED** [review]

**Comment by yangby-cryptape** [comment]
bors merge=doitian,quake

**Comment by yangby-cryptape** [comment]
bors retry

**Pull request merged**